resources/auth/controller.js

```diff
@@ -129,3 +128,55 @@ exports.viewProfile = async (req, res) => {
     res.status(500).json({ message: `Error loading profile ${error.message}` });
   }
 };
+
+exports.login = async (req, res) => {
+  try {
+    const {
+      user,
+      body: { password },
+    } = req;
+
+    /* 1st is user submitted password. 2nd is hashed stored password */
+    const isPasswordValid = bcrypt.compareSync(password, user.password);
+    if (isPasswordValid) {
+      delete user.password;
+      const token = generateToken(user);
+
+      res.status(200).json({
+        message: `Welcome. You're logged in!`,
+        data: { token, user },
+      });
+    } else {
+      res.status(401).json({ message: 'Invalid credentials' });
+    }
+  } catch (error) {
+    res.status(500).json({ message: `Failed to log user in` });
+  }
+};
+
+exports.updatePassword = async (req, res) => {
+  try {
+    const { oldPassword, newPassword, confirmPassword } = req.body;
+    const { user } = req;
+
+    const isOldPasswordValid = bcrypt.compareSync(oldPassword, user.password);
+
+    if (isOldPasswordValid) {
+      delete user.password;
+      const token = generateToken(user);
+
+      res.status(200).json({
+        message: `Welcome. You're logged in!`,
```

Comment:
if (isOldPasswordValid) {
      delete user.password;
      const token = generateToken(user);

      res.status(200).json({
        message: `Welcome. You're logged in!`,